### PR TITLE
Add support for uring probes

### PR DESCRIPTION
**Pull request by talex5**
Closes #51. /cc @patricoferris

### CODE REVIEW

File: lib/uring/dune [hunk 1]
```diff
@@ -12,7 +12,14 @@
...
 (rule
  (targets config.ml)
- (action (run ./include/discover.exe)))
+ (deps
+   include/liburing.h
+   include/liburing/io_uring.h
+   include/liburing.h
+   include/liburing/io_uring.h
+   include/liburing/barrier.h
+   include/liburing/compat.h)
```
Comment (bikallem) [inline lib/uring/dune:21]: TIL :+1:

File: lib/uring/include/discover.ml [hunk 2]
```diff
@@ -2,49 +2,122 @@ module C = Configurator.V1
...
-    C.C_define.import c ~c_flags:["-D_GNU_SOURCE"]
-      ~includes:["fcntl.h"; "poll.h"; "sys/uio.h"; "linux/time_types.h"]
-      C.C_define.Type.[
-          "POLLIN", Int;
-          "POLLOUT", Int;
-          "POLLERR", Int;
-          "POLLHUP", Int;
+      let defs =
+        C.C_define.import c ~c_flags:["-D_GNU_SOURCE"; "-I"; Filename.concat (Sys.getcwd ()) "include"]
+          ~includes:["fcntl.h"; "poll.h"; "sys/uio.h"; "linux/time_types.h"; "liburing.h"]
```
Comment (bikallem) [inline lib/uring/include/discover.ml:7]: We can remove "linux/time_types.h" from here if we can successfully include "liburing.h". liburing defines `__kernel_timespce` if not defined already - https://github.com/axboe/liburing/blob/master/configure#L446-L453.

File: lib/uring/include/discover.ml [hunk 3]
```diff
@@ -2,49 +2,122 @@ module C = Configurator.V1
...
+                    | "sizeof(struct __kernel_timespec)" -> "sizeof_kernel_timespec"
+                    | nm -> nm 
+                  in
+                  Printf.sprintf "let %s = 0x%x" (String.lowercase_ascii name) v
+                | _ -> assert false
+          )
+      in
+      let ops =
+        C.C_define.import c ~c_flags:["-D_GNU_SOURCE"; "-I"; Filename.concat (Sys.getcwd ()) "include"]
+          ~includes:["fcntl.h"; "poll.h"; "sys/uio.h"; "linux/time_types.h"; "liburing.h"]
```
Comment (bikallem) [inline lib/uring/include/discover.ml:53]: Same as above comment. `linux/time_types.h` is not needed if `liburing.h` can be included. Unless of course some other symbols defined in `linux/time_types.h` is needed.

File: lib/uring/include/discover.ml (hunk 2 above)
Comment (talex5) [reply lib/uring/include/discover.ml]: OK, I've removed it now.

File: lib/uring/include/discover.ml (hunk 3 above)
Comment (talex5) [reply lib/uring/include/discover.ml]: I've removed everything except `liburing.h` here.

**Pull request merged**